core/DataAccess/Model.php

```diff
@@ -713,54 +713,16 @@ public function startArchive($invalidation)
         $table = Common::prefixTable('archive_invalidations');
 
         // set archive value to in progress if not set already
-        $statement = Db::query("UPDATE `$table` SET `status` = ?, ts_started = NOW() WHERE idinvalidation = ? AND status = ?", [
+        $statement = Db::query("UPDATE `$table` SET `status` = ?, `processing_host` = ?, `process_id` = ?, `ts_started` = NOW() WHERE `idinvalidation` = ? AND `status` = ?", [
             ArchiveInvalidator::INVALIDATION_STATUS_IN_PROGRESS,
+            gethostname() ?: null,
+            Common::getProcessId(),
             $invalidation['idinvalidation'],
             ArchiveInvalidator::INVALIDATION_STATUS_QUEUED,
         ]);
 
-        if ($statement->rowCount() > 0) { // if we updated, then we've marked the archive as started
-            return true;
-        }
-
-        // archive was not originally started or was started within the expected time, so we assume it's ongoing and another process
-        // (on this machine or another) is actively archiving it.
-        $archiveFailureRecoveryTimeout = GeneralConfig::getConfigValue('archive_failure_recovery_timeout', $invalidation['idsite']);
-        if (
-            empty($invalidation['ts_started'])
-            || $invalidation['ts_started'] > Date::now()->subSeconds($archiveFailureRecoveryTimeout)->getTimestamp()
```

Comment:
This check never worked properly as `ts_archived` is provided from the database as datetime like `2025-01-01 12:00:00`. Comparing that this way with a timestamp always results in `true`, causing the code below the check to never get executed.
As we are resetting stale invalidations in the begin of the archiving process anyway, that shouldn't be required here anyway.